<launch>
  <!-- Basic args -->
		



  <arg name="mav_name"    default="firefly" />
  <arg name="world_name"  default="test_city_RRTC" />

  <!-- Spawnthe MAV -->
  <arg name="x" default="18.0" />
  <arg name="y" default="-18.0" />
  <arg name="z" default="0.1" />


 <!-- x_goal -->
  <arg name="goal_x"   default="-10" />
  <arg name="goal_y"   default="-30" />
  <arg name="goal_z"   default="0.1" />
  <arg name="goal_yaw" default="0.0" />

  <!-- Gazebo world -->
  <env name="GAZEBO_MODEL_PATH"    value="${GAZEBO_MODEL_PATH}:$(find rotors_gazebo)/models"/>
  <env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find rotors_gazebo)/models"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find rotors_gazebo)/worlds/$(arg world_name).world" />
    <arg name="paused"     value="true" />
    <arg name="gui"        value="true" />
    <arg name="verbose"    value="false" />
    <arg name="debug"      value="false" />
  </include>


  <group ns="$(arg mav_name)">


    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name"           value="$(arg mav_name)" />
      <arg name="model"              value="$(find rotors_description)/urdf/mav_generic_odometry_sensor.gazebo" />
      <arg name="enable_logging"     value="false" />
      <arg name="enable_ground_truth" value="true" />
      <arg name="log_file"           value="$(arg mav_name)" />
      <!-- pass spawn pose -->
      <arg name="x" value="$(arg x)" />
      <arg name="y" value="$(arg y)" />
      <arg name="z" value="$(arg z)" />
    </include>

    <!-- Lee controller -->
    <node name="lee_position_controller_node"
          pkg="rotors_control"
          type="lee_position_controller_node"
          output="screen">
      <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_$(arg mav_name).yaml" />
      <rosparam command="load" file="$(find rotors_gazebo)/resource/$(arg mav_name).yaml" />
      <remap from="odometry" to="odometry_sensor1/odometry" />
    </node>

    <!-- RRTC planner node -->
    <node name="RRTC_planner"
          pkg="uav_RRTC_planner"
          type="RRTC_planner_node.py"
          output="screen">
      <param name="mav_name"  value="$(arg mav_name)" />
      <param name="goal_x"    value="$(arg goal_x)" />
      <param name="goal_y"    value="$(arg goal_y)" />
      <param name="goal_z"    value="$(arg goal_z)" />
      <param name="goal_yaw"  value="$(arg goal_yaw)" />
    </node>

    <node name="robot_state_publisher"
          pkg="robot_state_publisher"
          type="robot_state_publisher" />
    <node name="joint_state_publisher"
          pkg="joint_state_publisher"
          type="joint_state_publisher" />
  </group>
</launch>

